File: leetcodecpp/lintcode/SixFour.cpp
//
// Created by John on 2020/4/11.
//

#include "vector"
using namespace std;

class Solution{
public:
    void mergeSortedArray(vector<int> A, int m, vector<int> B, int n) {
        int i = m - 1;
        int j = n - 1;
        int k = m + n - 1;
        while (i >= 0 && j >= 0) {
            if (A[i] > B[j]) {
                A[k--] = A[i--];
            } else {
                A[k--] = B[j--];
            }
        }
        while (j >= 0) {
            A[k--] = B[j--];
        }
    }
};

File: java/src/main/java/com/lintcode/SixFour.java
package com.lintcode;

/**
 * @author: John
 * @date: 2020-04-11 10:00
 * @description: 64
 **/
public class SixFour {
    public void mergeSortedArray(int[] A, int m, int[] B, int n) {
        int i = m - 1, j = n - 1, index = m + n - 1;
        while (i >= 0 && j >= 0) {
            if (A[i] > B[j]) {
                A[index--] = A[i--];
            } else {
                A[index--] = B[j--];
            }
        }

        while (i >= 0) {
            A[index--] = A[i--];
        }

        while (j >= 0) {
            A[index--] = B[j--];
        }
    }
}
